api/backend/ecs/service_manager.go

```diff
@@ -120,12 +120,28 @@ func (this *ECSServiceManager) DeleteService(environmentID, serviceID string) er
 	ecsServiceID := id.L0ServiceID(serviceID).ECSServiceID()
 	desiredCount := int64(0)
 
-	if err := this.ECS.UpdateService(ecsEnvironmentID.String(), ecsServiceID.String(), nil, &desiredCount); err != nil {
-		return err
+	service, err := this.GetService(environmentID, serviceID)
+        if err != nil {
+                return  err
+        }
+
+        taskARNs := []*string{}
+        for _, deployment := range service.Deployments {
+                arns, err := getTaskARNs(this.ECS, ecsEnvironmentID, stringp(deployment.DeploymentID))
+                if err != nil {
+                        return err
+                }
+
+                taskARNs = append(taskARNs, arns...)
+        }
+
+	for _, arn := range taskARNs{
+		if err := this.ECS.StopTask(ecsEnvironmentID.String(), "Service deleted by User", *arn); err != nil{
+			log.Warnf("Stop Task for Service '%s' had error: %v", serviceID, err)
+		}
 	}
 
-	log.Debugf("Waiting for service to stop")
-	if err := this.waitUntilServiceStopped(ecsEnvironmentID, ecsServiceID); err != nil {
```

Comment:
To expand on my point above, my concern is mostly relating to services that "do things" when they receive a graceful shutdown interrupt, i.e. write final log entries, set permissions or locks on external resources, etc. 

To that end, I wonder if it would be worth exposing two forms of `delete`: the old one that uses `waitUntilServiceStopped` and this new one that is more forceful.